ci/jenkins/test-mc.sh

```diff
@@ -296,7 +297,7 @@ function deliver_antrea_multicluster {
     chmod -R g-w build/images/ovs
     chmod -R g-w build/images/base
 
-    DOCKER_REGISTRY="${DOCKER_REGISTRY}" ./hack/build-antrea-linux-all.sh --pull
+    DOCKER_REGISTRY="${DOCKER_REGISTRY}" DOCKER_BUILDKIT=1 ./hack/build-antrea-linux-all.sh --pull
```

Comment:
Given that `./hack/build-antrea-linux-all.sh` assumes that BuildKit is enabled, we could just set `DOCKER_BUILDKIT=1` in the script directly, to avoid all these individual changes?

Reply:
BTW, I have been thinking of using `docker buildx build` directly as I believe it offers more command-line options (and better caching support), so it would make sense to update Docker everywhere.

Reply:
> BTW, I have been thinking of using `docker buildx build` directly as I believe it offers more command-line options (and better caching support), so it would make sense to update Docker everywhere.

Can I add buildx support in a seperate PR once I have upgraded docker on all testbeds? I can include this target in the issue of https://github.com/antrea-io/antrea/pull/5937#discussion_r1472622955.